### ISSUE TITLE
[WindowsError] [Error 32] The process cannot access the file because it is being used by another process:

### ISSUE THREAD

**Issue by monkeez** [sdispater/poetry#45]
After using the custom installer to install Poetry for Python 2.7, when I try to add a dependency to a project I get this Windows Error:
```
$ poetry add pendulum -vvv

Using version ^1.4 for pendulum

Updating dependencies
:   0: Starting resolution (2018-04-14 13:55:48.344000)
:   0: Requested dependencies: ['pendulum (^1.4)', 'pytest (^3.5)']
Resolving dependencies
:   0: Getting dependencies for pendulum-1.4.4
:   0: Getting dependencies for pendulum-1.4.3
:   0: Getting dependencies for pendulum-1.4.2
:   0: Getting dependencies for pendulum-1.4.1
:   0: Getting dependencies for pendulum-1.4.0
:   0: Creating possibility state for pendulum (^1.4) (5 remaining)
:   1: Attempting to activate [pendulum-1.4.4]
:   1: Activated pendulum at [pendulum-1.4.4]
:   1: Requiring nested dependencies (pytzdata (>= 2018.3.0.0), tzlocal (>= 1.5.0.0, < 2.0.0.0), python-dateutil (>= 2.6.0.0, < 3.0.0.0))
:   1: Getting dependencies for pytest-3.5.0
:   0: Creating possibility state for pytest (^3.5) (1 remaining)
:   2: Attempting to activate [pytest-3.5.0]
:   2: Activated pytest at [pytest-3.5.0]
:   2: Requiring nested dependencies (py (>= 1.5.0.0), six (>= 1.10.0.0), attrs (>= 17.4.0.0), more-itertools (>= 4.0.0.0), pluggy (< 0.7.0.0, >= 0.5.0.0), funcsigs (*), colorama (*))
:   2: Getting dependencies for pytzdata-2018.4

:   0: Finished resolution (2 steps) in 1.026 seconds

                                                                                                                                                                
  [WindowsError]                                                                                                                                                
  [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\mitch\\appdata\\local\\temp\\tmp_2aucm\\pytzdata-2018.4.tar.gz'
                                                                                                                                                                

Exception trace:
 c:\python27\lib\site-packages\poetry\_vendor\cleo\application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 c:\python27\lib\site-packages\poetry\console\application.py in do_run() at line 74
   return super(Application, self).do_run(i, o)
 c:\python27\lib\site-packages\poetry\_vendor\cleo\application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 c:\python27\lib\site-packages\poetry\console\commands\command.py in run() at line 22
   return super(BaseCommand, self).run(i, o)
 c:\python27\lib\site-packages\poetry\_vendor\cleo\commands\base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 c:\python27\lib\site-packages\poetry\_vendor\cleo\commands\command.py in execute() at line 107
   return self.handle()
 c:\python27\lib\site-packages\poetry\console\commands\add.py in handle() at line 87
   status = installer.run()
 c:\python27\lib\site-packages\poetry\installation\installer.py in run() at line 73
   self._do_install(local_repo)
 c:\python27\lib\site-packages\poetry\installation\installer.py in _do_install() at line 171
   ops = solver.solve(request, fixed=fixed)
 c:\python27\lib\site-packages\poetry\puzzle\solver.py in solve() at line 41
   graph = resolver.resolve(requested, base=base)
 c:\python27\lib\site-packages\poetry\mixology\resolver.py in resolve() at line 39
   base
 c:\python27\lib\site-packages\poetry\mixology\resolution.py in resolve() at line 93
   self._process_topmost_state()
 c:\python27\lib\site-packages\poetry\mixology\resolution.py in _process_topmost_state() at line 158
   self._attempt_to_activate()
 c:\python27\lib\site-packages\poetry\mixology\resolution.py in _attempt_to_activate() at line 775
   self._activate_new_spec()
 c:\python27\lib\site-packages\poetry\mixology\resolution.py in _activate_new_spec() at line 821
   self._require_nested_dependencies_for(self.possibility)
 c:\python27\lib\site-packages\poetry\mixology\resolution.py in _require_nested_dependencies_for() at line 851
   len(nested_dependencies) > 0
 c:\python27\lib\site-packages\poetry\mixology\resolution.py in _push_state_for_requirements() at line 884
   possibilities = self._possibilities_for_requirement(new_requirement)
 c:\python27\lib\site-packages\poetry\mixology\resolution.py in _possibilities_for_requirement() at line 908
   self._provider.search_for(requirement)
 c:\python27\lib\site-packages\poetry\mixology\resolution.py in _group_possibilities() at line 942
   dependencies = self._provider.dependencies_for(possibility)
 c:\python27\lib\site-packages\poetry\puzzle\provider.py in dependencies_for() at line 207
   complete_package = self._pool.package(package.name, package.version)
 c:\python27\lib\site-packages\poetry\repositories\pool.py in package() at line 62
   package = repository.package(name, version)
 c:\python27\lib\site-packages\poetry\repositories\pypi_repository.py in package() at line 120
   release_info = self.get_release_info(name, version)
 c:\python27\lib\site-packages\poetry\repositories\pypi_repository.py in get_release_info() at line 234
   lambda: self._get_release_info(name, version)
 c:\python27\lib\site-packages\poetry\_vendor\cachy\repository.py in remember_forever() at line 174
   val = value(callback)
 c:\python27\lib\site-packages\poetry\_vendor\cachy\helpers.py in value() at line 6
   return val()
 c:\python27\lib\site-packages\poetry\repositories\pypi_repository.py in <lambda>() at line 234
   lambda: self._get_release_info(name, version)
 c:\python27\lib\site-packages\poetry\repositories\pypi_repository.py in _get_release_info() at line 302
   requires_dist = self._get_requires_dist_from_urls(urls)
 c:\python27\lib\site-packages\poetry\repositories\pypi_repository.py in _get_requires_dist_from_urls() at line 322
   return self._get_requires_dist_from_sdist(urls['sdist'])
 c:\python27\lib\site-packages\poetry\repositories\pypi_repository.py in _get_requires_dist_from_sdist() at line 385
   return
 c:\python27\lib\contextlib.py in __exit__() at line 24
   self.gen.next()
 c:\python27\lib\site-packages\poetry\utils\helpers.py in temporary_directory() at line 30
   shutil.rmtree(name)
 c:\python27\lib\shutil.py in rmtree() at line 266
   onerror(os.remove, fullname, sys.exc_info())
 c:\python27\lib\shutil.py in rmtree() at line 264
   os.remove(fullname)

add [-D|--dev] [--optional] [--dry-run] [--] <name> (<name>)...
```

**Comment by sdispater** [comment]
@monkeez It should be fixed on the latest version (`0.8.2`).

You just have to update using `poetry self:update`.

**Comment by sdispater** [comment]
@monkeez As of version `0.8.0`, a fallback mechanism has been added in the dependency resolution process which will be slower (and sometimes much slower). This is to ensure all the dependencies are picked up. you can see more details [here](https://poetry.eustace.io/blog/poetry-0-8-0-is-out.html#fallback-mechanism-for-missing-dependencies) or [here](https://poetry.eustace.io/docs/basic-usage/#specifying-dependencies)

For now, there is no way around it unfortunately but you can disable it if you want, at the risk of having missing dependencies:

```bash
poetry config settings.pypi.fallback false
```

Note, that the releases information will be cached so further dependency resolution will be much faster.

Regarding, skipped dependencies, this is normal since `poetry` will keep information about every dependencies, even those not compatible with the current system.

However, in this case, some should not be skipped, but it is due to the fact that library maintainers b are putting anything in the platform field (like `Independent` for `pytz`). I will try to improve this but I can't guarantee that there will no longer be issues since the Python ecosystem can sometimes be a mess.

**Comment by monkeez** [comment]
Is there any reason why it would have skipped pytest though? pytest was in the development dependencies.

When I ran `poetry install` for a second time it installed pytest and all its dependencies. I'm not sure why it would work the second time.

**Comment by sdispater** [comment]
I am not sure either since the information is the same in both cases.

Anyway, I will rollback the change I made for platforms to avoid this issue for the time being while I think about a better solution to this.

**Comment by monkeez** [comment]
Thanks! I'll close this issue though as I can now add packages on Windows.

**Comment by sdispater** [comment]
@monkeez I just released a new version (`0.8.3`) which should fix the platform issue.
